client/lib/domains/email-forwarding/index.js

```diff
@@ -23,6 +25,17 @@ function validateField( { name, value } ) {
 	}
 }
 
+function validateDuplicatedForward( { name, value }, existingForwards ) {
+	switch ( name ) {
+		case 'mailbox':
+			return existingForwards?.filter( ( t ) => t.mailbox === value ).length > 1
+				? [ 'Repeated' ]
+				: [];
+		default:
+			return [];
+	}
```

Comment:
I think we should be able to simplify this block of code with:

```suggestion
	if ( name !== 'mailbox') {
		return [];
	}
	
	return existingForwards?.some( ( forward ) => forward.mailbox === value ) ? [ 'Repeated' ] : [];
```